packages/discovery-react-components/src/components/DocumentPreview/__tests__/DocumentPreview.test.tsx

```diff
@@ -28,91 +74,265 @@ describe('DocumentPreview', () => {
+        .mockImplementation(() => Promise.resolve(dummyResponse));
+      jest
+        .spyOn(searchClient, 'getComponentSettings')
+        .mockImplementation(() => Promise.resolve(dummyResponse));
+
+      const selectedResult = {
+        document: omit(pdfDoc, 'extracted_metadata.text_mappings'),
+        element: null,
+        elementType: null
+      };
+      const results = {
+        matching_results: 1,
+        results: [selectedResult]
+      };
+
+      act(() => {
+        ({ getByText } = render(
+          <DiscoverySearch
+            searchClient={searchClient}
+            projectId={'PROJECT_ID'}
+            overrideSearchResults={results}
+            overrideSelectedResult={selectedResult}
+          >
+            <DocumentPreview />
+          </DiscoverySearch>
+        ));
+      });
+      getByText('On 22 December 2008 ART EFFECTS LIMITED', { exact: false });
+    });
   });
 
-  it('renders document with html-like text', () => {
-    const htmlTextDoc = {
-      id: '1234567890',
-      extracted_metadata: {
-        filename: 'i_am_a_file'
-      },
-      text: ['Example text <text> <username> <password> more text afterwards']
-    };
+  // All of these currently fail since DocumentPreview defaults to HtmlView if a HTML field is present.
+  // Also fails because SimpleDocument will not show a JSON document without a passage highlight.
+  describe.skip('JSON files', () => {
+    it('renders text field for JSON files', async () => {
+      act(() => {
+        ({ getByText } = render(<DocumentPreview document={jsonDoc} />));
+      });
+      getByText('Hello, I am a text field.');
+    });
 
-    let getByText: NonNullable<any>;
+    it('renders an overwritten "body" field for JSON files', async () => {
+      context = {
+        componentSettings: {
+          fields_shown: {
+            body: {
+              field: 'body_field'
+            }
+          }
+        }
+      };
+      act(() => {
+        ({ getByText } = render(
+          wrapWithContext(<DocumentPreview document={jsonDoc} />, api, context)
+        ));
+      });
+      getByText('Display text from the specified "body" field.');
+    });
 
-    act(() => {
-      ({ getByText } = render(<DocumentPreview document={htmlTextDoc} />));
+    it('renders text from passage field, with passage highlighted', async () => {
+      const highlight = {
+        passage_text: 'Hello',
+        start_offset: 0,
+        end_offset: 4,
+        field: 'text'
+      };
+
+      act(() => {
+        ({ findAllByTestId } = render(
+          <DocumentPreview document={jsonDoc} highlight={highlight} />
+        ));
+      });
+
+      const highlights = await findAllByTestId('field-rect');
+      expect(highlights.length).toEqual(1);
+      const highlightContainer = highlights[0].parentElement;
+      expect(highlightContainer.getAttribute('data-field-type')).toEqual('passage');
+      expect(highlightContainer.getAttribute('data-field-id')).toEqual('0');
```

Comment:
Since these checks are common enough in this file, you should refactor them into a helper function.